modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -921,4 +818,133 @@ public int getSubscriptionCount( String queueName){
         }
     }
 
+    /**
+     * Method to display a list of messages when browsed
+     */
+    private CompositeData[] getDisplayableMetaData(List<AndesMessageMetadata> metadataList) throws MBeanException{
+        List<CompositeData> compositeDataList = new ArrayList<>();
+        try {
+            for (AndesMessageMetadata andesMessageMetadata : metadataList) {
```

Comment:
Divided into three methods by the PR https://github.com/wso2/andes/pull/403/files